### PR TITLE
feat: enhance group generation and introduce identity for connections

### PR DESCRIPTION
**Pull request by paulkr**

### CODE REVIEW

File: integrationos-api/src/logic/connection.rs [hunk 1]
```diff
@@ -163,20 +176,27 @@ pub async fn create_connection(
         }
     };
 
+    let group = Uuid::new_v4().to_string().replace('-', "");
+    let namespace = "default".to_string();
+    let name = match access.environment {
+        Environment::Test => format!("{} sandbox account", connection_config.name),
```
Comment (sagoez) [inline integrationos-api/src/logic/connection.rs:182]:
```suggestion
        Environment::Test | Environment::Development => format!("{} sandbox account", connection_config.name),
```

File: integrationos-api/src/logic/connection.rs [hunk 2]
```diff
@@ -163,20 +176,27 @@ pub async fn create_connection(
...
+        Environment::Development => format!("{} sandbox account", connection_config.name),
+        Environment::Live => format!("{} production account", connection_config.name),
```
Comment (sagoez) [inline integrationos-api/src/logic/connection.rs:184]:
```suggestion
        Environment::Live | Environment::Production => format!("{} production account", connection_config.name),
```

File: integrationos-api/src/logic/connection.rs [hunk 3]
```diff
@@ -163,20 +176,27 @@ pub async fn create_connection(
...
+        Environment::Production => format!("{} production account", connection_config.name),
```
Comment (sagoez) [inline integrationos-api/src/logic/connection.rs:185]:
```suggestion
```

File: integrationos-api/src/logic/connection.rs [hunk 4]
```diff
@@ -163,20 +176,27 @@ pub async fn create_connection(
         }
     };
 
+    let group = Uuid::new_v4().to_string().replace('-', "");
+    let namespace = "default".to_string();
+    let name = match access.environment {
+        Environment::Test => format!("{} sandbox account", connection_config.name),
+        Environment::Development => format!("{} sandbox account", connection_config.name),
```
Comment (sagoez) [inline integrationos-api/src/logic/connection.rs:183]:
```suggestion
```

File: integrationos-api/src/logic/oauth.rs [hunk 5]
```diff
@@ -189,17 +192,25 @@ async fn oauth_handler(
         })?;
 
     let conn_definition = get_conn_definition(&state, &payload.connection_definition_id).await?;
+    let group = Uuid::new_v4().to_string().replace('-', "");
+    let namespace = "default".to_string();
+    let name = match user_event_access.environment {
+        Environment::Test => format!("{} sandbox account", conn_definition.name),
```
Comment (sagoez) [inline integrationos-api/src/logic/oauth.rs:198]: DITTO. Optionally if this is repeated perhaps we could put this in a single place?

File: integrationos-api/tests/api_tests/test_server/mod.rs [hunk 6]
```diff
@@ -347,7 +347,7 @@ impl TestServer {
         &mut self,
         environment: Environment,
     ) -> (SanitizedConnection, ConnectionModelDefinition) {
-        let (key, access_key) = match environment {
+        let (key, _access_key) = match environment {
```
Comment (sagoez) [inline integrationos-api/tests/api_tests/test_server/mod.rs:350]:
```suggestion
        let (key, _) = match environment {
```

**Review by sagoez — CHANGES_REQUESTED** [review]

**Review by sagoez — APPROVED** [review]

**Review by sagoez — APPROVED** [review]

**Review by moekatib — APPROVED** [review]

**Pull request merged**